### DISCUSSION TITLE
如何动态修改自定义组建的props

### DISCUSSION THREAD

**Discussion by gdc1998** [alibaba/formily#3504] [Q&A 2.x] [answered]
form.setFieldState("zdy", (field) => {
            field.componentProps.data = 123
          });

https://formilyjs.org/zh-CN/guide/advanced/controlled#%E6%9C%80%E4%BD%B3%E5%AE%9E%E8%B7%B5
两种方式自定义拿不到最新的值

**Comment by frehaiku** [comment, marked as answer]
提供下复现demo

**Comment by gdc1998** [comment]
已解决。useMemo 导致
